src/main.rs

```diff
@@ -85,6 +103,25 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         });
     }
 
+    let (shutdown_tx, mut shutdown_rx) = mpsc::channel::<()>(1);
+
+    tokio::spawn(async move {
+        let service = MySnapchainService::default();
+
+        let resp = Server::builder()
+            .add_service(SnapchainServiceServer::new(service))
+            .serve(grpc_socket_addr)
+            .await;
+
+        let msg = "grpc server stopped";
+        match resp {
+            Ok(()) => error!(msg),
+            Err(e) => error!(error = ?e, "{}", msg),
+        }
+
+        shutdown_tx.send(()).await.ok();
```

Comment:
When will this be hit?

Reply:
I guess for any reason the grpc server stops running. I'm not sure the full set of cases, but one common one is when the grpc server address/port is already in use.